scripts/generate-dotnet.sh

```diff
@@ -1,14 +1,58 @@
 #!/bin/sh
 
-./scripts/install-tools.sh
+set -ex
+
+# TODO(kfcampbell): reduce duplication between this script and scripts/generate-go.sh
+if [ "$#" -lt 1 ] || [ "$#" -gt 2 ]; then
+    echo "Usage: $0 <platform> [version]"
+    echo "platform: must be one of 'ghes', 'ghec', or 'dotcom'"
+    echo "version: a currently supported version of GitHub Enterprise Server, required only for 'ghes' and omitted otherwise"
+    exit 1
+fi
+
+PLATFORM=$1
+VERSION=${2:-""} # version defaults to an empty string if not provided
+
+if [ "$PLATFORM" != "ghes" ] && [ "$PLATFORM" != "ghec" ] && [ "$PLATFORM" != "dotcom" ]; then
+    echo "Invalid platform: must be one of 'ghes', 'ghec', or 'dotcom'."
+    exit 1
+fi
 
-go run schemas/main.go --schema-next=false
+if [ "$PLATFORM" = "ghes" ]; then
+	# Note that this logic only validates that the version is a positive number
+    if [ "$VERSION" = "" ] || ! echo "$VERSION" | grep -Eq '^[0-9]+(\.[0-9]+)*$'; then
```

Comment:
This may be overkill; another option here would be removing version checking altogether and then erroring out in the schema download if the version doesn't exist in the `github/rest-api-description` repo.

Reply:
Possibly.  Doing so might help us avoid the whole ✅ on double digit version number parts. But I am pretty sure the expression above will catch that.  Were you thinking if the file fetch ends up in a 404 then log and move on?

Reply:
Yep, exactly. That would happen [here](https://github.com/octokit/source-generator/blob/main/schemas/main.go#L94-L98) without any additional effort on our part.